packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2330,6 +2330,33 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     }
   }
 
+  /// Creates the selection overlay if it is null.
+  ///
+  /// Returns `true` if the overlay was created, and `false` if it already
+  /// existed.
+  bool _createSelectionOverlayIfNeeded() {
+     if (_selectionOverlay == null) {
```

Comment:
Nit: Maybe reverse the logic of this method so that it is an early return.  So if it `!= null` then return false first.

Reply:
Good point, that's more readable. Fixed.